app/src/main/java/com/example/mynewsblog/ui/fragments/NewsFragment.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.mynewsblog.ui.fragments
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.mynewsblog.ui.adapters.NewsListAdapter
+import com.example.mynewsblog.data.NewsRepository
+import com.example.mynewsblog.data.model.NewsArticleModel
+import com.example.mynewsblog.databinding.FragmentNewsBinding
+import com.example.mynewsblog.ui.RecyclerViewSpacingItems
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class NewsFragment : Fragment() {
+
+    private lateinit var binding: FragmentNewsBinding
+    private var adapter: NewsListAdapter? = null
+
+    private var newsList = ArrayList<NewsArticleModel>()
+
+    @Inject
+    lateinit var newsRepository: NewsRepository
+
+    override fun onCreateView(inflater: LayoutInflater,container: ViewGroup?,savedInstanceState: Bundle?): View {
+
+        binding = FragmentNewsBinding.inflate(inflater, container, false)
+
+        adapter = NewsListAdapter(newsList,requireContext())
+
+        fetchNewsArticles()
+
+        binding.rvNews.adapter = adapter
+        binding.rvNews.layoutManager = LinearLayoutManager(context)
+
+        val spacingItems = RecyclerViewSpacingItems(95)
+        binding.rvNews.addItemDecoration(spacingItems)
+
+        return binding.root
+    }
+
+    private fun fetchNewsArticles() {
+
+        lifecycleScope.launch {
+            try {
```

Comment:
wrong approach

Reply:
use a viewModel and observer changes inside your view/fragment/actiivty

read about mvvm and observer pattern use kotlin flow along with sealed classes